src/Shared/FileUtilities.cs

```diff
@@ -1031,27 +1031,72 @@ internal static string MakeRelative(string basePath, string path)
+            string[] splitBase = fullBase.Split(new char[] { Path.DirectorySeparatorChar }, StringSplitOptions.RemoveEmptyEntries);
+            string[] splitPath = fullPath.Split(new char[] { Path.DirectorySeparatorChar }, StringSplitOptions.RemoveEmptyEntries);
 
-            Uri pathUri = CreateUriFromPath(path);
+            ErrorUtilities.VerifyThrow(splitPath.Length > 0, "Cannot call MakeRelative on a path of only slashes.");
 
-            if (!pathUri.IsAbsoluteUri)
+            // On a mac, the path could start with any number of slashes and still be valid. We have to check them all.
+            int indexOfFirstNonSlashChar = 0;
+            while (path[indexOfFirstNonSlashChar] == Path.DirectorySeparatorChar)
             {
-                // the path is already a relative url, we will just normalize it...
-                pathUri = new Uri(baseUri, pathUri);
+                indexOfFirstNonSlashChar++;
+            }
+            if (path.IndexOf(splitPath[0]) != indexOfFirstNonSlashChar)
+            {
+                // path was already relative so just return it
+                return FixFilePath(path);
             }
 
-            Uri relativeUri = baseUri.MakeRelativeUri(pathUri);
-            string relativePath = Uri.UnescapeDataString(relativeUri.IsAbsoluteUri ? relativeUri.LocalPath : relativeUri.ToString());
+            int baseI = 0;
+            int pathI = 0;
+            while (true)
+            {
+                if (baseI == splitBase.Length)
+                {
+                    if (pathI == splitPath.Length)
+                    {
+                        return ".";
+                    }
+                    break;
+                }
+                else if (pathI == splitPath.Length)
+                {
+                    break;
+                }
+                else if (splitBase[baseI].Equals(splitPath[pathI], PathComparison))
+                {
+                    baseI++;
+                    pathI++;
+                }
+                else
+                {
+                    break;
+                }
+            }
 
-            string result = relativePath.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+            StringBuilder sb = StringBuilderCache.Acquire();
 
-            return result;
+            // If the paths have no component in common, the only valid relative path is the full path.
+            if (baseI == 0)
+            {
+                return fullPath;
+            }
+            while (baseI < splitBase.Length)
+            {
+                sb.Append("..").Append(Path.DirectorySeparatorChar);
+                baseI++;
+            }
+            while (pathI < splitPath.Length)
+            {
+                sb.Append(splitPath[pathI]).Append(Path.DirectorySeparatorChar);
+                pathI++;
+            }
+            sb.Remove(sb.Length - 1, 1);
```

Comment:
nit: I would expect
```suggestion
            sb.Length = sb.Length - 1;
```
to be slightly faster and also easier to read.